tremor-connectors/src/impls/http/server.rs

```diff
@@ -230,48 +226,73 @@ impl Source for HttpServerSource {
         // Server task - this is the main receive loop for http server instances
         self.server_task = Some(spawn_task(ctx.clone(), async move {
             if let Some(server_config) = tls_server_config {
-                let make_service = make_service_fn(move |_conn: &tls::Stream| {
-                    // We have to clone the context to share it with each invocation of
-                    // `make_service`. If your data doesn't implement `Clone` consider using
-                    // an `std::sync::Arc`.
-                    let server_context = server_context.clone();
+                let incoming = TcpListener::bind(addr).await?;
 
-                    async move {
-                        // Create a `Service` for responding to the request.
-                        let service =
-                            service_fn(move |req| handle_request(server_context.clone(), req));
+                let acceptor = TlsAcceptor::from(server_config);
 
-                        // Return the service to hyper.
-                        anyhow::Ok(service)
-                    }
-                });
-                let incoming = AddrIncoming::bind(&addr)?;
-                let listener = hyper::Server::builder(tls::Acceptor::new(server_config, incoming))
-                    .serve(make_service);
-                listener.await?;
-            } else {
-                let make_service = make_service_fn(move |_conn: &AddrStream| {
+                loop {
                     // We have to clone the context to share it with each invocation of
                     // `make_service`. If your data doesn't implement `Clone` consider using
                     // an `std::sync::Arc`.
                     let server_context = server_context.clone();
 
-                    // Create a `Service` for responding to the request.
-                    let service =
-                        service_fn(move |req| handle_request(server_context.clone(), req));
+                    let (tcp_stream, _remote_addr) = incoming.accept().await?;
+
+                    let tls_acceptor = acceptor.clone();
+                    tokio::spawn(async move {
+                        let service = service_fn(move |req: Request<Incoming>| {
+                            handle_request(server_context.clone(), req)
+                        });
 
-                    // Return the service to hyper.
-                    async move { Ok::<_, Infallible>(service) }
-                });
-                let listener = hyper::Server::bind(&addr).serve(make_service);
+                        let tls_stream = match tls_acceptor.accept(tcp_stream).await {
+                            Ok(tls_stream) => tls_stream,
+                            Err(err) => {
+                                eprintln!("failed to perform tls handshake: {err:#}");
```

Comment:
let's use error so our logging framework catches this
```suggestion
                                error!("failed to perform tls handshake: {err:#}");
```

Reply:
I clearly wasn't thinking when eprintln slipped